src/autoTune.ml

```diff
@@ -375,6 +376,15 @@ class octagonVariableVisitor(varMap, globals) = object
         DoChildren
       )
     | Lval ((Var info),_) when not (isGoblintStub info) ->  handle varMap 1 globals (Some (`Right info)) ; SkipChildren
+    | UnOp (LNot, BinOp (op, e1,e2, (TInt _)),_) when isComparison op -> (* TODO: Should be generalized *)
+      handle varMap 5 globals (
+        match extractVar e1, extractVar e2 with
+        | Some a, Some b -> Some (`Left (a,b))
+        | Some a, None
+        | None, Some a -> if isConstant e1 then Some (`Right a) else None
```

Comment:
Ok, next try :smiley: : I know that is is done similarly in the `extractOctagonVars` part a little bit above, but why do we drop `a` when `e1` is constant? Should this not be symmetric, and `e2` be checked? Is the check providing anything more then extractVar already does anyway?

Reply:
If I were to guess, it would make sense to be symmetric. The asymmetry could be an accidental result of trying to collapse both `match` cases into one right-hand-side.